tests/unit/test_req.py

```diff
@@ -636,3 +636,25 @@ def test_mismatched_versions(caplog, tmpdir):
         'Requested simplewheel==2.0, '
         'but installing version 1.0'
     )
+
+
+class TestRequirementTracker(object):
+    @pytest.fixture(autouse=True)
+    def blank_env(self):
+        with patch.dict(os.environ, clear=True):
+            yield
+
+    def test_no_side_effects_on_init(self):
+        RequirementTracker()
+        assert 'PIP_REQ_TRACKER' not in os.environ
+
+    def test_keeps_original_environ(self, tmpdir):
+        os.environ['PIP_REQ_TRACKER'] = str(tmpdir)
+        with RequirementTracker():
+            assert os.environ['PIP_REQ_TRACKER'] == str(tmpdir)
+        assert os.environ['PIP_REQ_TRACKER'] == str(tmpdir)
+
+    def test_undoes_environ_set(self):
```

Comment:
How about `test_context_manager__without_PIP_REQ_TRACKER`?

Reply:
then it's not clear what the test is doing, requiring a code comment -- I'd rather keep as much noise out of the test as possible -- your suggested name doesn't help indicate what the test is demonstrating

Reply:
It tests entering and exiting, so I think the name should reflect that, with a focus on the `PIP_REQ_TRACKER`, and similarly for the other test.